provider/haproxy/scripts/haproxy_reload

```diff
@@ -4,19 +4,19 @@ set -e
 reload_haproxy(){
     # start rsyslog if not running 
     if [[ $(pgrep rsyslog | wc -l) != 1 ]]; then
-        rm -f /var/run/rsyslogd.pid
+        rm -f /run/rsyslogd.pid
         rsyslogd -f /etc/haproxy/rsyslogd.conf > /dev/null
     fi
 
-    if [ ! -f /var/run/haproxy.pid ] || [ $2 == "start" ]; then
-        if haproxy -p /var/run/haproxy.pid -f $1; then
+    if [ ! -f /run/haproxy.pid ] || [ $2 == "start" ]; then
+        if service haproxy start; then
             return 0
         else
             return 1
         fi
     fi
-    # restart service
-    if haproxy -p /var/run/haproxy.pid -f $1 -sf $(cat /var/run/haproxy.pid); then
+    # reload service
+    if service haproxy reload; then
```

Comment:
looks like haproxy reload does the same thing as the line 19 in the old code:
```
haproxy_reload()
{
        check_haproxy_config

        $HAPROXY -f "$CONFIG" -p $PIDFILE -sf $(cat $PIDFILE) -D $EXTRAOPTS \
                || return 2
        return 0
}
```
so it is fine to replace it with reload. Can you check though if you can pass custom config file to the ```service haproxy <command>``` ($1 option that gets dropped with your PR)? Even though we use the default file location today, I'd like to maintain the flexibility of specifying custom config location